src/components/app/data/hooks/useDefaultSearchFilters.ts

```diff
@@ -0,0 +1,119 @@
+import { useContext, useEffect, useMemo } from 'react';
+import { SearchContext, setRefinementAction, SHOW_ALL_NAME } from '@edx/frontend-enterprise-catalog-search';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+import useSearchCatalogs from './useSearchCatalogs';
+import useAlgoliaSearch from './useAlgoliaSearch';
+import { FilterBuilder } from '../../../FilterBuilder';
+
+interface SearchContextValue {
+  refinements: Record<string, any>;
+  dispatch: (action: any) => void;
+}
+
+interface QueryByCatalogArgs {
+  searchCatalogs: string[];
+  enterpriseCustomer: EnterpriseCustomer;
+  showAllRefinement: boolean;
+}
+
+interface QueryByCatalogQueryArgs {
+  catalogUuidsToCatalogQueryUuids: Record<string, string>;
+  searchCatalogs: string[];
+}
+
+/**
+ * Constructs a filter string scoped to either all enterprise catalogs or a specific subset,
+ * depending on whether `showAllRefinement` is enabled or the search catalog list is empty.
+ *
+ * @param searchCatalogs - Catalog UUIDs to refine the search by
+ * @param enterpriseCustomer - The current enterprise customer context
+ * @param showAllRefinement - Whether to bypass catalog filters and show all
+ * @returns A composed Algolia filter string
+ */
+const queryByCatalog = ({
+  searchCatalogs,
+  enterpriseCustomer,
+  showAllRefinement,
+}: QueryByCatalogArgs): string => {
```

Comment:
nit: TS should be able to infer the string type from the returned `.build()`, typed upstream, without explicitly typing this function.